app/status.go

```diff
@@ -137,7 +137,9 @@ func GetUserStatusesByIds(userIds []string) ([]*model.Status, *model.AppError) {
 		}
 	}
 	for _, userId := range missingUserIds {
-		statusMap = append(statusMap, &model.Status{UserId: userId, Status: "offline"})
+		if result := <-Srv.Store.User().Get(userId); result.Err == nil {
+			statusMap = append(statusMap, &model.Status{UserId: userId, Status: "offline"})
```

Comment:
We already hit the database above to try and get the missing user ids. We can either leave it as it was and return offline for ids not found in the cache or in the DB, or remove this loop and not return anything for the missing ids. I lean towards the second option since that's how the other ids endpoints work